### ISSUE TITLE
Include strategy

### ISSUE THREAD

**Issue by HazardyKnusperkeks** [hanickadot/compile-time-regular-expressions#265]
Hi,

I'm starting to use this library, but have some issues with the include strategy. I'd like to have a proper folder for the inclusion, something like: `#include <ctre/ctre.hpp>` like https://github.com/fmtlib/fmt, boost, or https://github.com/mpusz/units.

If one included some headers directly, we basically have this already. I wanted only `ctre::match` so I `#include <ctre/wrapper.hpp>` because it's defined there. But here lies another problem, the headers are not self containing. So I have to ask what is the difference between `include/ctre.hpp` and `single-header/ctre.hpp` I always get everything, right?

**Comment by hanickadot** [comment]
Always include just <ctre.hpp> it's not designed for including only a selected header. Set your include folder to include/ or single-header/ result should be identical.

**Comment by Alcaro** [comment]
The difference between the two ctre.hpp is whether it's a single header, nothing else. The latter just replaces every #include with the target file's contents, flattening CTRE to a single file and letting you control how you want to arrange your directories.

**Comment by HazardyKnusperkeks** [comment]
> Always include just <ctre.hpp> it's not designed for including only a selected header. Set your include folder to include/ or single-header/ result should be identical.

Yeah, it'd be great, if there was some sort of hierarchy. Of course it's your choice whether to grant such a request or not.

> The difference between the two ctre.hpp is whether it's a single header, nothing else. The latter just replaces every #include with the target file's contents, flattening CTRE to a single file and letting you control how you want to arrange your directories.

So basically only for those who want to copy the `single-header` variant somewhere else? Did not think about such use case.
